core/src/main/java/com/linecorp/armeria/client/proxy/ProxyConfig.java

```diff
@@ -201,6 +201,12 @@ public static ProxyConfig direct() {
     @Nullable
     public abstract InetSocketAddress proxyAddress();
 
+    /**
+     * Returns a new proxy address instance that respects DNS TTL.
+     * @param newProxyAddress the inet socket address
+     */
+    public abstract ProxyConfig withNewProxyAddress(InetSocketAddress newProxyAddress);
```

Comment:
nit; I believe that currently only resolved `InetSocketAddress` can be used to create a `ProxyConfig`.
Can you add some integration tests to confirm the current code works correctly?

Reply:
I added two integration test codes for this feature. 
Due to restrictions on mocking, narrow `Access specifier`, I couldn't use a `spy` instance to validate it or inject `DNSResolver` as well. 
That's why I made a new class called `DNSResolverFacadeUtils` class 😅 .

As @minwoox mentioned, we don't need to check if `InetSocketAddress` is resolved, since it will be resolved when `HttpClientDelegate` calls `execute()`. After removing `checkArgument(...)`, I was finally able to write some integration test cases... 😅 

When you have time, please take another look.  🙇‍♂️